app/database_etl/test_adjustment_handler/test_adjustment.py

```diff
@@ -270,3 +63,122 @@ def run_on_test_set(model_code: str = testadj_model_code, model_name: str = 'tes
     records_df['ind_eval_type'], records_df['adj_prev_ci_lower'], records_df['adj_prev_ci_upper'] = \
         zip(*records_df.apply(lambda row: testadjHandler.get_adjusted_estimate(row), axis=1))
     return records_df
+
+
+def add_test_adjustments(df: pd.DataFrame) -> pd.DataFrame:
+    # Query record ids in our database
+    with db_session() as session:
+        total_db_records = session.query(DashboardSource.serum_pos_prevalence,
+                                         DashboardSource.test_adj,
+                                         DashboardSource.sensitivity,
+                                         DashboardSource.specificity,
+                                         DashboardSource.test_type,
+                                         DashboardSource.denominator_value,
+                                         DashboardSource.adj_prevalence,
+                                         ResearchSource.ind_se,
+                                         ResearchSource.ind_sp,
+                                         ResearchSource.ind_se_n,
+                                         ResearchSource.ind_sp_n,
+                                         ResearchSource.se_n,
+                                         ResearchSource.sp_n,
+                                         ResearchSource.test_validation,
+                                         ResearchSource.airtable_record_id) \
+            .join(ResearchSource, ResearchSource.source_id == DashboardSource.source_id, isouter=True).all()
+        total_db_records = [q._asdict() for q in total_db_records]
+        total_db_records = pd.DataFrame(data=total_db_records)
+
+    # Concat old and new records and fillna with 0 (NaN and None become 0 so it is standardized)
+    diff = pd.concat([df, total_db_records])
+    diff.fillna(0, inplace=True)
+
+    # Convert numeric cols to float (some of these come out of airtable as strings so need to standardize types)
+    float_cols = ['ind_se', 'ind_sp', 'ind_se_n', 'ind_sp_n', 'se_n', 'sp_n', 'sensitivity', 'specificity',
+                  'denominator_value', 'serum_pos_prevalence']
+    diff[float_cols] = diff[float_cols].astype(float)
+
+    # Round float columns to a consistent number of decimal places to ensure consistent float comparisons
+    diff[float_cols] = diff[float_cols].round(5)
+
+    # Drop duplicates based on these cols
+    duplicate_cols = ['airtable_record_id', 'test_adj', 'ind_se', 'ind_sp', 'ind_se_n', 'ind_sp_n',
+                      'se_n', 'sp_n', 'sensitivity', 'specificity', 'test_validation', 'test_type', 'denominator_value',
+                      'serum_pos_prevalence']
+    diff = diff.drop_duplicates(subset=duplicate_cols, keep=False)
+
+    # Get all unique airtable_record_ids that are new/have been modified
+    new_airtable_record_ids = diff['airtable_record_id'].unique()
+
+    # Add all unique airtable_record_ids for which test adjustment was unsuccessful
```

Comment:
Are we ever using this commented out code anymore? If not can we get rid, it's an eyesore.

If do need this logic to rerun test adjustment for data refresh purposes, I would personally prefer keeping it out of this function (maybe make a separate script/utility to do this)

Reply:
took it out. we won't be doing this in the etl